### ISSUE TITLE
Upgrading from 0.9.3 to 0.11 throws error when terraform apply is executed due to unauthorized error

### ISSUE THREAD

**Issue by PradeepSingh1988** [hashicorp/terraform#17326]
### Terraform Version
0.11.0

### Terraform Configuration Files

```hcl
resource "aws_instance" "test_instance" {
  ami = "ami-xyz"
  key_name = "key-name-xyz"
  instance_type = "t2.medium"
  iam_instance_profile = "profile1"
  tags {
    Name        = "Pradeep_Test"
    Environment = "Pradeep_Test"
    Owner       = "Pradeep"
  }
  subnet_id = "subnet-pqr"
  # ALL to resources under same VPC and inbound 443 for others
  vpc_security_group_ids = ["sg-xyz"]
}

```

Instance Profile is something like below:
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "sts:AssumeRole",
      "Principal": {
        "Service": "ec2.amazonaws.com"
      },
      "Effect": "Allow",
      "Sid": ""
    }
  ]
}
```

### Debug Output

```
9.0_x4: <Response><Errors><Error><Code>UnauthorizedOperation</Code><Message>You are not authorized to perform this operation.</Message></Error></Errors><RequestID>******</RequestID></Response>
2018-02-13T05:58:23.049Z [DEBUG] plugin.terraform-provider-aws_v1.9.0_x4: 2018/02/13 05:58:23 [DEBUG] [aws-sdk-go] DEBUG: Validate Response ec2/ModifyNetworkInterfaceAttribute failed, not retrying, error UnauthorizedOperation: You are not authorized to perform this operation.
```

### Expected Behavior

Terraform should succeed.
### Actual Behavior

It failed with error.

**Comment by catsby** [comment]
Hello @PradeepSingh1988 –

Can you elaborate on what operation is taking place here? You mention upgrade, so I assume you've replaced v0.9.3 binary with newer 0.11.0 binary. What steps were taken then? Running `terraform plan`? If so, was there any plan output? More information can help use narrow things down.

Thanks!

**Comment by PradeepSingh1988** [comment]
Yes, I replaced terraform 0.9.3 binary with 0.11.0, earlier i did not need to add "ec2:ModifyNetworkInterfaceAttribute" in my instance profile, but after i upgraded to to 0.11.0 i need this because terraform is making a call to update the interface when it tries to do that for ```vpc_security_group_ids = ["sg-xyz"]```. I think earlier implementation of this call was different.
Currently i don't have the same setup, can we get any information from this.

**Comment by apparentlymart** [comment]
I believe we've seen this reported before... it changed as part of adding support for some new EC2 features  that thus required some additional operations.

Since it's an issue with the AWS provider I'm going to have our bot move the issue over there. However, I believe there is already an existing issue about this (which I was not able to find in a quick search) which the maintainers of that provider may choose to consolidate this into.
